cmd/collector/app/zipkin/http_handler_test.go

```diff
@@ -0,0 +1,283 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zipkin
+
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	jaegerClient "github.com/uber/jaeger-client-go"
+	zipkinTransport "github.com/uber/jaeger-client-go/transport/zipkin"
+	tchanThrift "github.com/uber/tchannel-go/thrift"
+
+	"github.com/uber/jaeger/thrift-gen/jaeger"
+	"github.com/uber/jaeger/thrift-gen/zipkincore"
+)
+
+var httpClient = &http.Client{Timeout: 2 * time.Second}
+
+type mockJaegerHandler struct {
+	err     error
+	mux     sync.Mutex
+	batches []*jaeger.Batch
+}
+
+type mockZipkinHandler struct {
+	err   error
+	mux   sync.Mutex
+	spans []*zipkincore.Span
+}
+
+func (p *mockZipkinHandler) SubmitZipkinBatch(ctx tchanThrift.Context, spans []*zipkincore.Span) ([]*zipkincore.Response, error) {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+	p.spans = append(p.spans, spans...)
+	return nil, p.err
+}
+
+func (p *mockZipkinHandler) getSpans() []*zipkincore.Span {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+	return p.spans
+}
+
+func initializeTestServer(err error) (*httptest.Server, *APIHandler) {
+	r := mux.NewRouter()
+	handler := NewAPIHandler(&mockZipkinHandler{err: err})
+	handler.RegisterRoutes(r)
+	return httptest.NewServer(r), handler
+}
+
+func TestFormatsViaClient(t *testing.T) {
+	server, handler := initializeTestServer(nil)
+	defer server.Close()
+
+	zipkinSender, err := zipkinTransport.NewHTTPTransport(
+		server.URL+`/api/v1/spans`,
+		zipkinTransport.HTTPBatchSize(1),
+	)
+	require.NoError(t, err)
+
+	tracer, closer := jaegerClient.NewTracer(
+		"test",
+		jaegerClient.NewConstSampler(true),
+		jaegerClient.NewRemoteReporter(zipkinSender),
+	)
+	defer closer.Close()
+
+	tracer.StartSpan("root").Finish()
+
+	deadline := time.Now().Add(2 * time.Second)
+	for {
+		if time.Now().After(deadline) {
+			t.Error("never received a span")
+			return
+		}
+		if want, have := 1, len(handler.zipkinSpansHandler.(*mockZipkinHandler).getSpans()); want != have {
+			time.Sleep(time.Millisecond)
+			continue
+		}
+		break
+	}
+
+	assert.Equal(t, 1, len(handler.zipkinSpansHandler.(*mockZipkinHandler).getSpans()))
+}
+
+func TestThriftFormat(t *testing.T) {
+	span := &zipkincore.Span{}
+	spans := []*zipkincore.Span{}
```

Comment:
given that in this test you don't actually use span, you can inline it like so:

`spans := []*zipkincore.Span{{}}`